File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/HotelModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HotelModel {
    private Long id;

    private String name;
    private String address;
    private Integer stars;
    private Double pricePerNight;
    private String roomType;
    private Integer capacity;
    private String services;
    private String location;

    private List<PackageHotelModel> packages;
}

File: frontend/tour-design-web/src/app/models/hotel.model.ts
// src/app/models/hotel.model.ts
import { PackageHotelModel } from './package-hotel.model';

export interface HotelModel {
  id: number;
  name: string;
  address?: string;
  stars?: number;
  pricePerNight?: number;
  roomType?: string;
  capacity?: number;
  services?: string;
  location?: string;

  packages?: PackageHotelModel[];
}
